go-controller/pkg/cluster/node.go

```diff
@@ -139,30 +139,29 @@ func (cluster *OvnClusterController) StartClusterNode(name string) error {
 
 	type readyFunc func(string, string) (bool, error)
 	var readyFuncs []readyFunc
-	var nodeAnnotations map[string]string
-	var postReady postReadyFn
 
-	// If gateway is enabled, get gateway annotations
-	if config.Gateway.Mode != config.GatewayModeDisabled {
-		nodeAnnotations, postReady, err = cluster.initGateway(node.Name, subnet.String())
-		if err != nil {
-			return err
-		}
-		readyFuncs = append(readyFuncs, GatewayReady)
+	// get gateway annotations
+	gwAnnotations, postReady, err := cluster.initGateway(node.Name, subnet.String())
+	if err != nil {
+		return err
 	}
+	readyFuncs = append(readyFuncs, GatewayReady)
 
 	// Get management port annotations
 	mgmtPortAnnotations, err := CreateManagementPort(node.Name, subnet, clusterSubnets)
 	if err != nil {
 		return err
 	}
-
 	readyFuncs = append(readyFuncs, ManagementPortReady)
 
-	// Combine mgmtPortAnnotations with any existing gwyAnnotations
+	// Combine mgmtPortAnnotations and gwAnnotations into nodeAnnotations
+	nodeAnnotations := make(map[string]interface{})
```

Comment:
@girishmg wait, why does this need to be `interface{}` rather than `string`? Annotations are strings only, so I don't think it makes sense to mkae it an interface.